daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/CompilerTest.scala

```diff
@@ -285,8 +297,9 @@ class CompilerTest extends AnyWordSpec with Matchers with Inside {
           )
 
           inside(evalSExpr(sexpr, getContract = Map(contractId1 -> versionedContract1))) {
-            case Right((SValue.SUnit, contractCache)) =>
+            case Right((SValue.SUnit, contractCache, disclosedContractKeys)) =>
               contractCache.keySet shouldBe Set(contractId1)
+              disclosedContractKeys.values.map(_.value).toSet shouldBe Set(contractId1)
```

Comment:
Should not we check there is not duplicate entries in  `disclosedContractKeys.values.map(_.value)` ? 
The question applies to other test cases where `disclosedContractKeys` is not empty.

```suggestion
              disclosedContractKeys.values.map(_.value).toList.sorted shouldBe List(contractId1).sorted
```